### DISCUSSION TITLE
How to test the sempy_labs.report package?

### DISCUSSION THREAD

**Discussion by dtiles** [microsoft/semantic-link-labs#187] [Q&A] [answered]
Hi to all,
I was interested to test in Fabric the sempy_labs.report package. I tried to create a ReportWrapper, but what I get as an error is 

> ValueError: 🔴 The ReportWrapper function requires the report to be in the PBIR format.See here for details: https://powerbi.microsoft.com/blog/power-bi-enhanced-report-format-pbir-in-power-bi-desktop-developer-mode-preview/

However, if I enable Power BI Project mode with the PBIR format...I can't publish to the service. I can't even import the PBIX converted from a PBIP with PBIR...so there is no way to test this package until the PBIR format goes in GA?

Best regards

Daniele Tiles

**Comment by m-kovalsky** [comment, marked as answer]
Hi Daniele,

Please follow the instructions within the link provided in the error message. You must:

* Enable the preview feature in Power BI desktop (then close/reopen)
* Configure your Fabric workspace to Git
* Save the Power BI Desktop file as a .pbip.
* Commit and push the change to Git (using VSCode)
* In the Fabric workspace, update the changes so that your workspace reflects what is in Git.

After completing this process, you will have a Power BI report in the service which is in the PBIR format. Note that the PBIR format is in preview so there are limitations. The ReportWrapper was designed with the future in mind (not the past) - that is why it requires PBIR (which will be the default format for reports when it goes GA).

**Reply by dtiles** [reply]
Hi,
thanks for the the answer. Unfortunately our clients haven't got Azure DevOps or GitHub, so we cannot test it.
We'll wait when it goes GA, hopefully it'll work  with published Power BI Projects.
Do you think it'll work also with Premium Capacities? Our clients haven't switched yet to Fabric Capacities

**Reply by m-kovalsky** [reply]
Fabric also works on Premium capacities (as long as it is enabled). Once PBIR goes GA, it should be much easier, yes.

**Comment by knlsarda13** [comment]
Hey @m-kovalsky ,
Once saving the file in PBIP format with the upgrade to PBIR option. I see the following files generated as follows:
<PBI Name>.Report
<PBI Name>.SemanticModel
<PBI Name>.pbip
<PBI Name>.pbix

Now, I have pushed all the files to my GIT, but the 5 step: "In the Fabric workspace, update the changes so that your workspace reflects what is in Git."

The files are not showing in the Updates section in Source Control, am I missing something here?
![image](https://github.com/user-attachments/assets/a2133260-35ea-4d01-b668-2241dbc58c7e)
